src/validators.test.ts

```diff
@@ -66,3 +68,102 @@ describe('length', () => {
     expect(got).toEqual(expectedError);
   });
 });
+
+// Pattern validator
+describe('pattern', () => {
+  it('returns error', () => {
+    const expectedError = `Invalid characters in input`;
+    expect(pattern(/^[a-b]/)('z')).toEqual(expectedError);
+  });
+  it('returns undefined', () => {
+    expect(pattern(/^[a-z0-9]*$/)('12ba')).toEqual(undefined);
+  });
+  it('considers errorMsg', () => {
+    const errorMsg = 'Custom error.';
+    expect(pattern(/^[a-z0-9]*$/, errorMsg)('abc_123')).toEqual(errorMsg);
+  });
+})
+
+// minlength validator
+describe('minlength', () => {
+  it('returns error', () => {
+    const expectedError = `Should be at least ${5} characters long`;
+    expect(minlength(5)("suku")).toEqual(expectedError);
+  });
+  it('returns undefined', () => {
+    expect(minlength(5)("summit")).toEqual(undefined);
+  });
+  it('considers errorMsg', () => {
+    const errorMsg = 'Custom error.';
+    expect(minlength(6, errorMsg)("suku")).toEqual(errorMsg);
+  });
+})
+
+// maxlength validator
+describe('maxlength', () => {
+  it('returns error', () => {
+    const length = 3
+    const expectedError = `Should be no longer than ${length} characters`;
+    expect(maxlength(length)("suku")).toEqual(expectedError);
+  });
+  it('returns undefined', () => {
+    expect(maxlength(6)("summit")).toEqual(undefined);
+  });
+  it('considers errorMsg', () => {
+    const errorMsg = 'Custom error.';
+    expect(maxlength(3, errorMsg)("suku")).toEqual(errorMsg);
+  });
+})
+
+// min validator
+describe('min', () => {
+  it('returns error', () => {
+    const digit = 5;
+    const expectedError = `Should be at least ${digit} digits long`;
+    expect(min(digit)(4037)).toEqual(expectedError);
```

Comment:
`min` does not check the length of a number but checks the value itself.

When we say `min(5)`, the value should be `>= 5`